src/Lean/Syntax.lean

```diff
@@ -305,6 +305,10 @@ def getRange? (stx : Syntax) (canonicalOnly := false) : Option String.Range :=
   | some start, some stop => some { start, stop }
   | _,          _         => none
 
+/-- Returns a synthetic Syntax which has the specified `String.Range`. -/
+def ofRange (range : String.Range) (canonical := true) : Lean.Syntax :=
```

Comment:
This just screams "workaround" to me, I'd really prefer it to be in a more specific place and namespace

Reply:
It is, kind of, but I do think it deserves to be centrally located. What it is working around is the fact that lots of server / info tree stuff uses `Syntax` where it really means `String.Range`, which causes many unnecessary optional unwraps and gives no good option for cases where the range being highlighted does not correspond to a single syntax. There are already a few places in lean core where the workaround of "make a null node with two `Syntax`es for the start and end bit" is used to graft a range together from pieces. I think it would be better to standardize on a single way of representing such dummy spans.

Reply:
Is this the only std use? https://github.com/leanprover/std4/blob/453fb67573767ca73eb4f08710c6cb352a40aaa5/Std/Tactic/TryThis.lean#L350
Isn't that just
```
stx := origSpan?.getD ref
```
?

Reply:
@Kha, I don't know the exact intended behaviour in the place you linked, but the code is not equivalent to `orginSpan?.getD ref`.

The code is using the range of `ref` if it has a range, and otherwise insisting that `origSpan?.isSome` and that it has a range, and using that (and then creating a dummy Syntax from that).

Reply:
Yes, I noticed that but found it unlikely to be intentional. Anyway, this function is used in the wild so I guess let's merge.

Reply:
This is paged out at the moment, but I thought quite carefully about that line and I'm quite sure the semantics there are intentional.